### PR TITLE
25 create checkpoint app

### PR DESCRIPTION
**Pull request by KarnaushenkoVO**
## TRELLO

* [Main Trello ticket](https://trello.com/c/B159UGFY/25-create-checkpoint-app)

## Code reviewers

- [x] @hrytskivr
- [x] @PanLeon
- [x] @yuriiZherebukh

### Second Level Review

- [ ] @byshovets
- [ ] @lhalam

## Summary of issue
Created checkpoint app for myTrip project

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: myTrip/checkpoint/views.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+""" This checkpoint module generates view for CRUD requests"""
+
+import json
+from django.http import JsonResponse, HttpResponse
+from django.views.generic import View
+from .models import  Checkpoint
```
Comment (hrytskivr) [inline myTrip/checkpoint/views.py:6]: add newline before imports from files

Comment (KarnaushenkoVO) [reply myTrip/checkpoint/views.py:6]: imports should not be there

File: myTrip/checkpoint/models.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+"""This module contains comment model class and basic functions"""
+
+from django.db import models
+from django.db.models import ProtectedError
+from django.core.exceptions import ObjectDoesNotExist, FieldError
+
```
Comment (lhalam) [inline myTrip/checkpoint/models.py:6]: delete 1 empty line

File: myTrip/checkpoint/models.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    Checkpoint
+    :argument id: int - autogenerated primary key
+    :argument longitude: float - 1-st position coordinate
+    :argument latitude: float - 2-nd position coordinate
+    :argument title: char - checkpoint's title
+    :argument description: text - description of checkpoint
+    :argument position_number: int - ordinal number of checkpoint
+    :argument source_url: url - url of the checkpoint's source
+    :argument trip: int - ToDo foreign key to Trip model
+    """
```
Comment (lhalam) [inline myTrip/checkpoint/models.py:19]: add empty line

File: myTrip/checkpoint/models.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            "latitude": self.latitude,
+            "title": self.title,
+            "description": self.description,
+            "source_url": self.source_url,
+            "position_number": self.position_number,
+            "trip": self.trip
+        }
+
+    @staticmethod
+    def create(data):
```
Comment (lhalam) [inline myTrip/checkpoint/models.py]: use all args name 
docstring for example:
```
"""
Get Like with given trip id
 Args:
    trip_id (int): trip id.
Returns:
    QuerySet<Like>: QuerySet of Like.
"""
```

File: myTrip/checkpoint/models.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    latitude = models.FloatField()
+    title = models.CharField(max_length=20)
+    description = models.TextField()
+    position_number = models.IntegerField()
+    source_url = models.URLField()
+    trip = models.IntegerField()
+
+    def to_dict(self):
+        """
+        Serialize model and choose fields
```
Comment (lhalam) [inline myTrip/checkpoint/models.py]: for example:
```
"""Convert model object to dictionary.
        Return:
            dict:
                {
                    'id': id,
                    'user_id': name,
                    ...
                }
        """
```

File: myTrip/checkpoint/models.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            return None
+        return new_object
+
+    @staticmethod
+    def get_by_id(checkpoint_id):
+        """
+        Return checkpoint objects with specific id or None if there is no object with such id
+        """
+        try:
+            return Checkpoint.objects.get(id=checkpoint_id)
```
Comment (lhalam) [inline myTrip/checkpoint/models.py]: use constructions
checkpoint = Checkpoint.objects.get(id=checkpoint_id)
return checkpoint

File: myTrip/checkpoint/models.py [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        """ Update field in checkpoint and save change"""
+        attributes = ['longitude', 'latitude', 'title', 'description',
+                      'position_number', 'source_url', 'trip']
+        for attribute in data.keys():
+            if attribute in attributes:
+                setattr(self, attribute, data[attribute])
+        self.save()
+        return True
+
+    def delete_checkpoint(self):
```
Comment (lhalam) [inline myTrip/checkpoint/models.py]: don't use class name in the method

**Review by yuriiZherebukh — APPROVED** [review]

**Pull request merged**